main.py

```
@@ -1,78 +1,69 @@
+            image_path,
+            font_path,
+            int(size),
+            position,
+            text,
+            align[0] if align else "left",
+            text_color,
+            highlighting_color,
+        )
+        return result_path
+    except Exception as e:
+        raise gr.Error(f"Processing failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)